model/src/main/kotlin/config/OrtConfiguration.kt

```diff
@@ -68,6 +68,24 @@ data class OrtConfiguration(
      */
     val enableRepositoryPackageConfigurations: Boolean = false,
 
+    /**
+     * A set with the names of environment variables that are passed to child processes created by ORT, e.g. when
+     * running external package managers or scanners. All variables not listed here are not passed through. This
+     * property aims to prevent that sensitive information from environment variables like passwords or service
+     * credentials is passed to external tools. As a default value, a larger number of well-known variables evaluated
+     * by the tools invoked by ORT is set. Users can choose to override this by redefining this property in their
+     * configuration file. In order to extend the set, the [additionalVariablesToPropagate] property should be used
+     * instead.
+     */
+    val variablesToPropagate: Set<String> = emptySet(),
+
+    /**
+     * A set with the names of additional environment variables that are passed to child processes created by ORT.
+     * Defining this property in a configuration file is the easiest way to extend the set of variables to pass
+     * through, since it is not necessary to redefine all the variables ORT takes into account by default.
+     */
+    val additionalVariablesToPropagate: Set<String> = emptySet(),
```

Comment:
I'd prefer to not need this. If above we'd set a `setOf(...)` default variables, if should be easy enough for the user to run `ort config --show-default`, copy over the default, append additional wanted variables, and set the combined collection as the new value.

Reply:
The mechanism has been completely reworked.